server/consumer.go

```diff
@@ -988,8 +1043,19 @@ func (o *consumer) setLeader(isLeader bool) {
 		}
 
 		// Update the group on the our starting sequence if we are starting but we skipped some in the stream.
-		if o.dseq == 1 && o.sseq > 1 {
-			o.updateSkipped()
+		// TODO(tp): check how we should approach this for multiple filters
+		if len(o.cfg.FilterSubjects) != 0 {
+			sort.Slice(o.msseq, func(i, j int) bool {
+				return o.msseq[j].sequence > o.msseq[i].sequence
+			})
+
+			if o.dseq == 1 && o.msseq != nil && o.msseq[0].sequence > 1 {
```

Comment:
Not sure if that's really an issue here, but checking `o.msseq != nil` does not prevent `o.msseq` from having `len == 0` (and therefore panicking). It would be safer to do:
```
if o.dseq == 1 && len(o.msseq) > 0 && o.msseq[0].sequence > 1
```

Reply:
that's a good point. fixed.